### ISSUE TITLE
Stop Warnings on External Modules Problems

### ISSUE THREAD

**Issue by jojomon42153** [pylint-dev/pylint#8962]
### Current problem

Starting to work with Pylint, and my big first problem with it is that when I inherit from an external class, some problems are in the way:
Example with implementation of coapthon3 from tanganelli:

```python3
from coapthon.resources.resource import Resource


class BasicResource(Resource):
    def __init__(self, name="BasicResource", coap_server=None):
        # Some init code here

    def render_GET(self, request):
        return self
```
Here it is:
Resource is an abstract class with MANY abstract methods (render_GET, render_POST, render_GET_separate, ...)
We can see here 2 problems (that explodes my score):
- C0103 invalid-name EACH time i override those ugly method names
- Dunno why but i have W0223 on vscode but not on pre-commit linting

For example on my whole file for C0103:
```shell
serial_coap/exampleresources.py:18:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:22:4: C0103: Method name "render_PUT" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:27:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:31:4: C0103: Method name "render_DELETE" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:43:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:47:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:59:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:63:4: C0103: Method name "render_PUT" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:67:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:73:4: C0103: Method name "render_DELETE" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:86:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:90:4: C0103: Method name "render_GET_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:95:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:99:4: C0103: Method name "render_POST_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:103:4: C0103: Method name "render_PUT" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:107:4: C0103: Method name "render_PUT_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:111:4: C0103: Method name "render_DELETE" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:115:4: C0103: Method name "render_DELETE_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:127:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:164:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:168:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:187:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:202:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:217:4: C0103: Method name "render_PUT" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:221:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:233:4: C0103: Method name "render_GET" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:237:4: C0103: Method name "render_POST" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:243:4: C0103: Method name "render_PUT" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:253:4: C0103: Method name "render_GET_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:260:4: C0103: Method name "render_POST_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:268:4: C0103: Method name "render_PUT_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:276:4: C0103: Method name "render_DELETE_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:288:4: C0103: Method name "render_GET_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:292:4: C0103: Method name "render_POST_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:296:4: C0103: Method name "render_PUT_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:300:4: C0103: Method name "render_DELETE_advanced" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:304:4: C0103: Method name "render_GET_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:311:4: C0103: Method name "render_POST_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:316:4: C0103: Method name "render_PUT_separate" doesn't conform to snake_case naming style (invalid-name)
serial_coap/exampleresources.py:321:4: C0103: Method name "render_DELETE_separate" doesn't conform to snake_case naming style (invalid-name)
```

### Desired solution

Natively, without user regex overriding manage of those problems that are external to the dev

**Comment by DanielNoord** [comment]
You can change the regex that is being used for `invalid-name`. 
See:
https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/invalid-name.html

**Comment by jojomon42153** [comment]
The requested feature was not a regex change as asked in the first message.
It was a natural understanding from pylint that the style is from an external library, and so don't warn at any moment.
If we have to override a very undigest regex each time we use an external library with other style, I personnally find it not user friendly.

**Comment by DanielNoord** [comment]
Although I see merit in that request it would be quite hard to figure out which project is "external". I don't think we want to start doing so as it is easy to introduce even more false positives.

**Comment by DanielNoord** [comment]
Yes! Or we will get a feature request to not consider something an external dependency if the name of the project is also in the name of the project for distributed projects.

I think the regex is the most flexible solution we have that can fit all use cases although I understand it is a bit hard to get right the first time.

**Comment by Pierre-Sassoulas** [comment]
I agree with Daniel about not detecting external dependencies, it feel like a full time project. We could however introduce a list of classes that have invalid-name in their api (for example by default it would have unitest.TestCase at least) and not raise invalid-name when another class inherit and override their api.
